src/test/java/io/unlaunch/engine/OperatorTest.java

```diff
@@ -27,6 +26,45 @@ public void testEquals(){
         Assert.assertEquals("Result EQ: true", true, result);
     }
 
+    @Test
+    public void testEqualsWrongDateFormat(){
+        /*
+        Assert.assertThrows(UnlaunchAttributeCastException.class,
+                () -> Operator.EQUALS.apply("1612314129000", new UnlaunchStringValue("invalidDate"), AttributeType.DATE)
+        ); */
+        try {
+            Operator.EQUALS.apply("1612314129000", new UnlaunchStringValue("invalidDate"), AttributeType.DATE);
+        } catch (UnlaunchAttributeCastException e) {
+            Assert.assertNotNull(e);
+        }
+    }
+
+    @Test
+    public void testEqualsWrongSetFormat(){
+        /*
+        Assert.assertThrows(UnlaunchAttributeCastException.class,
+                () -> Operator.EQUALS.apply("value", new UnlaunchBooleanValue(true), AttributeType.SET)
+        ); */
+        try {
+            Operator.EQUALS.apply("value", new UnlaunchBooleanValue(true), AttributeType.SET);
+        } catch (UnlaunchAttributeCastException e) {
+            Assert.assertNotNull(e);
+        }
+    }
+
+    @Test
+    public void testEqualsWrongNumberFormat(){
```

Comment:
If the exception is not thrown, the catch block won't be called and the test will pass. 
A better way is to add expected annotation to error. E.g.

@Test(expected = NullPointerException.class)
public void whenExceptionThrown_thenExpectationSatisfied() {
    String test = null;
    test.length();
}